src/main/java/com/uid2/admin/audit/QLDBAuditMiddleware.java

```diff
@@ -0,0 +1,134 @@
+package com.uid2.admin.audit;
+
+import com.amazon.ion.IonStruct;
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonSystemBuilder;
+import com.uid2.admin.auth.AdminUserProvider;
+import com.uid2.shared.auth.IAuthorizable;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.RoutingContext;
+import software.amazon.awssdk.services.qldbsession.QldbSessionClient;
+import software.amazon.qldb.QldbDriver;
+import software.amazon.qldb.RetryPolicy;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class QLDBAuditMiddleware implements AuditMiddleware{
+
+    public static final String LEDGER_NAME = "audit-logs";
+    public static final String TABLE_NAME = "logs";
+    private static final IonSystem ionSys = IonSystemBuilder.standard().build();
+    private static final QldbDriver qldbDriver = QldbDriver.builder()
```

Comment:
don't you need username/password to access QLDB?

Reply:
Reads username/pw from file on the computer/server